js/load.js

```diff
@@ -48,3 +51,6 @@ window.pkp = Object.assign(PkpLoad, {
 // Required by the URN plugin
 window.pkp.Vue.component('field-text', FieldText);
 window.pkp.Vue.component('catalog-list-panel', CatalogListPanel);
+
+// DOI
+window.pkp.Vue.component('DoiListPanel', DoiListPanel);
```

Comment:
Rather than adding this globally, it is preferred to add a `Container` component that defines the list panel as a dependency. First you'd create `components/Container/DoiPage.vue`:

```
import DoiListPanel from '@/components/ListPanel/doi/DoiListPanelOJS.vue';

export default {
	components: {
		DoiListPanel
	}
}
```

Then in `load.js` you load that instead:

```
import DoiPage from '@/components/Container/DoiPage.vue';

...

window.pkp = Object.assign(PkpLoad, {
	controllers: {
		...,
		DoiPage
	}
};
```

At the moment there's not much benefit to this, but eventually our build package will grow too large and we'll split it. As long as each `Page` component defines its dependencies, it will be easy for us to split things up into smaller packages. In other words, eventually we'd only load `DoiListPanel` on pages where it was used.

Reply:
Oh yeah, and then in the doi page handler you would add:

```
$templateManager->assign('pageComponent', 'DoiPage');
```

Reply:
I've poked around with this a bit and can't quite wrap my head around how this would fit in with/replace what's already there. Maybe this is something we could look at together?